widgets/htdocs/alphafold/dataFetchers.js

```diff
@@ -0,0 +1,145 @@
+import { getHexColor } from "./colorHelpers.js";
+import { proteinFeatureTypesSet } from "./proteinFeaturesControlPanel.js";
+
+const commonOptions = {
+  headers: {
+    'Content-Type': 'application/json'
+  },
+};
+
+/**
+ * @param {Object} params - Function parameters.
+ * @param {string} params.rootUrl - Root url of the api.
+ * @param {string} params.enspId - Ensembl identifier of the protein molecule.
+*/
+export const fetchAlphaFoldId = async (params) => {
+  const { rootUrl: apiRoot, enspId } = params;
+  const url = `${apiRoot}/overlap/translation/${enspId}?feature=protein_feature;type=alphafold`;
+  const alphafoldFeatures = await fetch(url, commonOptions)
+    .then(response => response.json()); // should be an array of one feature
+  const { id: alphaFoldId } = alphafoldFeatures[0];
+
+  // Note that the alphafold id will end in the name of the chain (e.g. "AF-Q9S745-F1.A").
+  // The chain has to be discarded when passed to Molstar
+  return alphaFoldId.split('.').shift();
+};
+
+/**
+ * @param {Object} params - Function parameters.
+ * @param {string} params.rootUrl - Root url of the api.
+ * @param {string} params.enspId - Ensembl identifier of the protein molecule.
+ */
+export const fetchExons = async (params) => {
+  const { rootUrl: apiRoot, enspId } = params;
+  const url = `${apiRoot}/overlap/translation/${enspId}?feature=translation_exon`;
+  const exons = await fetch(url, commonOptions)
+    .then(response => response.json());
```

Comment:
have you checked what happens when the response is empty?

Reply:
We do not know if there is going to be an empty response that is going to break the code below. Better if we write this code bit defensive..  As you mentioned @azangru  what if there are genes without protein family or variation data?

Reply:
I think the best we can do is check whether there is data for variants and for protein families before rendering the respective panels on the right-hand side. `dataFetchers.js` should be fine; if there's an empty array in the response, the fetchers will just return an empty array.

Reply:
I added a check that makes sure that the fetched data for variants and protein features actually has some items in it.

Reply:
Thanks. Looks good.